code/src/components/TaskList.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react'
+import { useSelector } from 'react-redux'
+import { TaskInput } from './TaskInput.js';
+import { TaskItem } from './TaskItem.js';
+import { TaskSummary } from './TaskSummary.js';
+import { ClearButton } from './ClearButton'
+
+export const TaskList = () => {
+  const [category, setCategory] = useState('')
+  const list = useSelector((store) => {
+    if (!category) {
+      console.log('no category!')
+      return store.tasks.list.items
+    } else {
+      console.log('category chosen!', category)
+      return store.tasks.list.items.filter((item) => item.category === category)
+    }
+  })
```

Comment:
If I understood your question about the categories correctly, you could add an action and a selector to get/change the current category.

`useSelector((store) => store.tasks.currentCategory)`

And you could change the category when adding an item to that category.